### PR TITLE
fix(interfaces): Updating Combo Chart Option type

### PR DESCRIPTION
**Pull request by YuriBarssi**
### Updates
- Updating the ComboChartOptions to match the object in the Demo (it was giving an error of incompatible objects)
image

### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/carbon-design-system/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

### CODE REVIEW

File: packages/core/src/interfaces/charts.ts [hunk 1]
```diff
@@ -320,13 +320,12 @@ export interface RadarChartOptions extends BaseChartOptions {
...
  */
 export interface ComboChartOptions extends AxisChartOptions {
-	comboChartTypes?: {
-		[ChartTypes.LINE]?: [],
-		[ChartTypes.STACKED_BAR]?: [],
-		[ChartTypes.GROUPED_BAR]?: [],
-		[ChartTypes.SIMPLE_BAR]?: [],
-		[ChartTypes.SCATTER]?: []
-	};
+	experimental?: boolean;
```
Comment (natashadecoste) [inline packages/core/src/interfaces/charts.ts:323]:
```suggestion
```

we don't need the experimental flag - we are just using this so the storybook demos have the experimental notification.

**Review by natashadecoste — CHANGES_REQUESTED** [review]
Thank you for this PR 🥳

File: packages/core/src/interfaces/charts.ts (hunk 1 above)
Comment (YuriBarssi) [reply packages/core/src/interfaces/charts.ts:323]: @natashadecoste Sure thing! This means I also do not need to define it in my type - then, as this is an 'storybook only property'?

File: packages/core/src/interfaces/charts.ts [hunk 2]
```diff
@@ -320,13 +320,11 @@ export interface RadarChartOptions extends BaseChartOptions {
...
 export interface ComboChartOptions extends AxisChartOptions {
-	comboChartTypes?: {
-		[ChartTypes.LINE]?: [],
-		[ChartTypes.STACKED_BAR]?: [],
-		[ChartTypes.GROUPED_BAR]?: [],
-		[ChartTypes.SIMPLE_BAR]?: [],
-		[ChartTypes.SCATTER]?: []
-	};
+	comboChartTypes?: Array<{
+		type?: ChartTypes;
```
Comment (natashadecoste) [inline packages/core/src/interfaces/charts.ts:324]:
```suggestion
		type: ChartTypes;
```

**Review by natashadecoste — CHANGES_REQUESTED** [review]
just one little fix and 👍 
`type` should be present to instantiate the component

**Comment by theiliad** [comment]
> just one little fix and 👍
> `type` should be present to instantiate the component

don't we also want the entire flag to be required? `comboChartTypes`?

File: packages/core/src/interfaces/charts.ts [hunk 3]
```diff
@@ -320,16 +326,13 @@ export interface RadarChartOptions extends BaseChartOptions {
...
  */
 export interface ComboChartOptions extends AxisChartOptions {
-	comboChartTypes?: {
-		[ChartTypes.LINE]?: [],
-		[ChartTypes.STACKED_BAR]?: [],
-		[ChartTypes.GROUPED_BAR]?: [],
-		[ChartTypes.SIMPLE_BAR]?: [],
-		[ChartTypes.SCATTER]?: []
-	};
+	comboChartTypes?: Array<{
```
Comment (theiliad) [inline packages/core/src/interfaces/charts.ts:329]:
```suggestion
	comboChartTypes: Array<{
```

**Review by theiliad — CHANGES_REQUESTED** [review]

**Comment by YuriBarssi** [comment]
@theiliad @natashadecoste changes addressed! Could I get another review 🔍 ? Thanks

**Review by theiliad — APPROVED** [review]

**Comment by YuriBarssi** [comment]
@theiliad @natashadecoste looks like the build is failing on lacking the tools lib. Is this a known issue? Is there something I need to do on my side? 
image

**Comment by theiliad** [comment]
> @theiliad @natashadecoste looks like the build is failing on resolving `@carbon/charts/tools`. Is this a known issue? Is there something I need to do on my side?
> image

yeah not sure why step demos are using `@carbon/charts/tools`, probably added with the new combo charts. Those should just be `Object.assign()`s. My #820 PR fixes that

File: packages/core/src/interfaces/charts.ts [hunk 4]
```diff
@@ -317,19 +317,16 @@ export interface RadarChartOptions extends BaseChartOptions {
...
 export interface ComboChartOptions extends AxisChartOptions {
-	comboChartTypes?: {
-		[ChartTypes.LINE]?: [],
-		[ChartTypes.STACKED_BAR]?: [],
-		[ChartTypes.GROUPED_BAR]?: [],
-		[ChartTypes.SIMPLE_BAR]?: [],
-		[ChartTypes.SCATTER]?: []
-	};
+	comboChartTypes: Array<{
+		type: ChartTypes;
```
Comment (natashadecoste) [inline packages/core/src/interfaces/charts.ts:324]:
```suggestion
		type: ChartTypes | any;
```

**Review by natashadecoste — CHANGES_REQUESTED** [review]

**Review by theiliad — APPROVED** [review]

**Comment by YuriBarssi** [comment]
@theiliad @natashadecoste I had to modify another file - it was complaining about the non-optional property type. Could you guys take another look? Thanks!!

**Review by theiliad — APPROVED** [review]

**Review by natashadecoste — APPROVED** [review]

**Pull request merged**